codec-http/src/main/java/io/netty/handler/codec/spdy/SpdyFrameDecoderExtendedDelegate.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2024 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.handler.codec.spdy;
+
+import io.netty.buffer.ByteBuf;
+
+/**
+ * {@link SpdyFrameDecoderDelegate} that also supports unknown frames.
+ */
+public interface SpdyFrameDecoderExtendedDelegate extends SpdyFrameDecoderDelegate {
```

Comment:
Elegant

Reply:
@He-Pin I am also wondering if you even have a need to implement this or do you just use `SpdyFrameCodec` ?

Reply:
I will extend `SpdyFrameCodec` as `ACCSSpdyFrameCodec` and hope there will be a method to create the unknown frame:

```java
SpdyUnknownFrame spdyUnknownFrame = new DefaultSpdyUnknownFrame(frameType, flags, payload);
```
can be done with 
```java
SpdyFrame spdyUnknownFrame = createUnknownFrame(frameType, flags, payload)
```

and then In my implementation, I will override the `createUnknownFrame` method with:
```java
public SpdyFrame createUnknownFrame(int frameType, byte flags, ByteBuf payload) {
        final byte[] data = ByteBufUtil.getBytes(payload);
        final CustFrame custFrame = newCustomFrame((short) frameType, flags, data);
        getChannelHandlerContext().fireChannelRead(custFrame);
}
```

With this, I can reduce another allocation of the `SpdyUnknownFrame` and then convert the ``SpdyUnknownFrame` to `CustFrame`.

I wanted to change our codec (ACCSFrame) to zero-copy around the ByteBuf which is currently around byte[], but that will require more work. 

So, I think the current shape is good enough but better with a  `createUnknownFrame` method, and the `SpdyFrameDecoderExtendedDelegate` will not be used in my codec for now.

Reply:
The current definition of `CustFrame` is something like:
```java
public final class CustFrame implements SpdyFrame {
	private final ControlFrameType type;
	private final byte flags;
	private final byte[] data; //TODO Can be done with ByteBuf after migrate to Netty
	private final String dataId;
```

Reply:
Our current usage of `SpdyFrameCodec` :
Where `AbstractSpdyFrameCodec` is actually a copy of `SpdyFrameCodec`.
```java
public final class ACCSSpdyFrameCodec extends AbstractSpdyFrameCodec
    implements ACCSSpdyFrameDecoderDelegate {

    public ACCSSpdyFrameCodec() {
        super(SpdyVersion.SPDY_3_1);
    }

    @Override
    protected ACCSSpdyFrameDecoder createDecoder(final SpdyVersion version,
                                                 final SpdyFrameDecoderDelegate delegate,
                                                 final int maxChunkSize) {
        return new ACCSSpdyFrameDecoder(this);
    }

    @Override
    protected ACCSSpdyFrameEncoder createEncoder(final SpdyVersion version) {
        return new ACCSSpdyFrameEncoder();
    }

    @Override
    public void write(final ChannelHandlerContext ctx,
                      final Object msg,
                      final ChannelPromise promise) throws Exception {
        if (msg instanceof CustFrame) {
            final CustFrame custFrame = (CustFrame) msg;
            final ByteBuf frame = getSpdyFrameEncoder().encodeCustomFrame(ctx.alloc(), custFrame);
            ctx.write(frame, promise);
        } else {
            super.write(ctx, msg, promise);
        }
    }

    @Override
    public void readUnknownFrame(final int frameType, final byte flags, final ByteBuf payload) {
        setRead(true);
        final byte[] data = ByteBufUtil.getBytes(payload);
        final CustFrame custFrame = newCustomFrame((short) frameType, flags, data);
        getChannelHandlerContext().fireChannelRead(custFrame);
    }


    private CustFrame newCustomFrame(final short controlFrameType, final byte flags, final byte[] data) {
        final ControlFrameType type = ControlFrameType.from(controlFrameType);
        return new CustFrame("", data, flags, type);
    }

    @Override
    public ACCSSpdyFrameDecoder getSpdyFrameDecoder() {
        return (ACCSSpdyFrameDecoder) super.getSpdyFrameDecoder();
    }

    @Override
    public ACCSSpdyFrameEncoder getSpdyFrameEncoder() {
        return (ACCSSpdyFrameEncoder) super.getSpdyFrameEncoder();
    }
}
```

After this released I will migrate to it.

Reply:
ok with this in mind I think we can just hide the interface completely... one sec

Reply:
Thank you.

Reply:
@He-Pin made the interface package-private... We will add the method directly to the old interface with a default method in the port to 4.2